src/core/public/mocks.ts

```diff
@@ -54,12 +55,13 @@ function createCoreSetupMock() {
 }
 
 function createCoreStartMock() {
-  const mock: MockedKeys<CoreStart> = {
+  const mock: MockedKeys<LegacyCoreStart> = {
```

Comment:
This mock incorrect. `injectedMetadata` is not exposed to the new platform plugin, but legacy plugins only. We decided to provide it for a while until ConfigService is done https://github.com/elastic/kibana/pull/47319

Reply:
So what should I do if I need to mock `injectedVars`?

Reply:
1 Revert this change.
2 Merge master into your branch. `getInjectedVar` was **temporarily** exposed from NP in https://github.com/elastic/kibana/pull/47319